rfcs/text/0000-matrix-service.md

```diff
@@ -0,0 +1,408 @@
+| Optimus           | `*zarr*`<br />`empty_drops_result.csv`                       | For Optimus pipelines, expression data is parsed from the pipeline generated `zarr` according to this [specification](https://github.com/HumanCellAtlas/skylab/blob/master/docs/matrix_format_spec.md) maintained by Secondary Analysis and Matrix teams. However, a large majority of the barcodes represented in the `zarr`do not represent genuine cells according to `empty_drops_result.csv`  (in one example bundle, only ~1% of rows are suggested to be real cells). To enable users to perform their own filtering on a reasonable set of barcodes while not delivering an abundance of low-quality data, a barcode is included in the index if the UMI count reported in `empty_drops_result.csv` is greater than 100, as entires with less are almost certainly artifactual. Additionally, users may still interact with these parameters via the `cell` table fields `barcode`, `total_umis` and `empty_drops_iscell`. | Within the index, cells are identified by a unique `cellkey` deterministically generated and managed by the Matrix Service. The current algorithm for generating an Optimus `cellkey` is the MD5 hash of the `bundle_uuid` and `barcode`. To ensure uniqueness and deterministic generation, the following assumptions must be true:<br /><br />1. A `bundle_uuid` must be preserved across updates and revisions to the bundle.<br />2. Barcodes are unique within a bundle.<br /><br />It should be noted that the event of a project reingestion does not maintain assumption (1) since bundle UUIDs are regenerated which results in generating a different `cellkey` for the same cell. The current response to this is to rebuild the index for the reingested project. |
+
+**Metadata**
+
+*Related tables:* `analysis`, `donor`, `specimen`, `cell_suspension`, `library_preparation`, `project`, `publication`, `contributor`
+
+The majority of metadata is parsed directly from metadata JSON files available in primary bundles. These fields are identifiable as fields with a valid _Canonical Metadata Names_ mapping in _Table 2.1_. The exceptions are "derived fields" such as `derived_organ_*` that require a degree of intellectual reasoning of the experiment to determine an accurate value. For example, the value of `derived_organ_ontology` may differ from the value of `organ_ontology`, which is read directly from `specimen_from_organism.organ.ontology`, as it is determined by the presence of organoids and/or cell lines in the experiment.
+
+Metadata entities are identified by their `*.provenance.document_id` (DIDs) which are consistently used as primary keys in the schema defined in *Table 2.1*. A similar set of assumptions applied to _Cell IDs_ described in *Table 2.2* apply to these IDs as well:
+
+1. DIDs are unique for distinct metadata entities.
+2. DIDs are preserved across all bundles updates to preserve referential integrity of metadata entities over time.
```

Comment:
The current plan for AUDR Analysis Updates will violate this assumption. On an analysis update, all the files will get different UUIDs apart from `analysis_process_0.json`.

Reply:
Resolved in [Slack](https://humancellatlas.slack.com/archives/C6SGU0R3J/p1574189870221900)